kolibri/core/tasks/exceptions.py

```diff
@@ -15,3 +15,14 @@ class JobNotFound(Exception):
 
 class JobNotRestartable(Exception):
     pass
+
+
+class ErrorSavedWithJob(Exception):
+    def __init__(self, message, prior_type):
+        if message is None or message == "":
+            message = "This job had an error before it was saved to the database."
+        self.prior_type = prior_type
+        super(ErrorSavedWithJob, self).__init__(message)
+
+    def __str__(self):
+        return "{} Exception type: {}".format(self.message, self.prior_type)
```

Comment:
I think even simpler would be just to add a rule for serialization exceptions to JSON.

It would could store the import path for the Exception class, and any relevant details (note the traceback is important here). Then the job can reserialize and create the Exception class from the details in the JSON object.